src/main/java/com/microsoft/sqlserver/jdbc/SQLServerXAResource.java

```diff
@@ -943,4 +945,62 @@ private static int nextResourceID() {
         return baseResourceID.incrementAndGet();
     }
 
+    private enum ResourceManagerFailure {
+        CONN_RESET("Connection reset");
+
+        private final String errString;
+
+        ResourceManagerFailure(String errString) {
+            this.errString = errString;
+        }
+
+        @Override
+        public String toString() {
+            return errString;
+        }
+
+        static ResourceManagerFailure fromString(String errString) {
+            for (ResourceManagerFailure resourceManagerFailure : ResourceManagerFailure.values()) {
+                if (errString.equalsIgnoreCase(resourceManagerFailure.toString())) {
+                    return resourceManagerFailure;
+                }
+            }
+            return null;
+        }
+    }
+
+    /**
+     * Check if the root exception of the throwable should be a XAER_RMFAIL exception
+     *
+     * @param throwable
+     *        The exception to check if the root cause should be a XAER_RMFAIL
+     *
+     * @return True if XAER_RMFAIL, otherwise false
+     */
+    private boolean isResourceManagerFailure(Throwable throwable) {
+        Throwable root = Util.getRootCause(throwable);
+
+        if (null == root) {
+            return false;
+        }
+
+        ResourceManagerFailure err = ResourceManagerFailure.fromString(root.getMessage());
+
+        if (null == err) {
+            return false;
+        }
+
+        // Add as needed here for future XAER_RMFAIL exceptions
+        if (root instanceof SocketException) {
+            switch (err) {
+                case CONN_RESET:
+                    return true;
+                default:
+                    return false;
+            }
```

Comment:
just return err == CONN_RESET

Reply:
or actually just
return (root instanceof SocketException) && err == CONN_RESET

Reply:
I this was made a switch instead of a simple return to make it easier to add future cases. Is it better to future proof, or just make the changes in the future when we eventually need to?